packages/truffle-contract/lib/utils.js

```diff
@@ -117,49 +141,77 @@ var Utils = {
   },
 
   // Extracts optional tx params from a list of fn arguments
-  getTxParams: function(args){
+  getTxParams: function(methodABI, args) {
     var constructor = this;
 
-    var tx_params =  {};
+    var expected_arg_count = methodABI ? methodABI.inputs.length : 0;
```

Comment:
Reworked this logic to be a bit more robust. The only time `methodABI` should be falsy is in the case of a call out to the fallback function.